### PR TITLE
Hash registry id

### PR DESCRIPTION
**Pull request by bricewang**
### Description

Hashes registry ID before making EVM call to `Registry.getAddressFor`.

### Tested

No tests added.

### Related issues

- Sister PR of https://github.com/celo-org/celo-monorepo/pull/657

### CODE REVIEW

**Comment by m-chrzan** [comment]
The registry hashes the input string, which is work that could be easily done off-chain.

I don't feel strongly about whether we should leave the old interface available as well. Accepting only hashes, as this PR does, leaves less clutter in the contract.

**Comment by asaj** [comment]
What do you mean by "on-chain"? We're either hashing here in the EVM or outside the EVM, but since the former is a precompile I would imagine it just calls the same code.

If we actually wanted efficiency gains we would only compute the hash of the identifier once

**Comment by m-chrzan** [comment]
> If we actually wanted efficiency gains we would only compute the hash of the identifier once

Oh yes, of course. We can totally do that, I'll add that change.

But the change in monorepo does already give us efficiency gains in any contract that calls out to other contracts, since in each spot like that we call `getAddressFor`, and there we do compute them once before deployment in `UsingRegistry`.

**Comment by asaj** [comment]
> But the change in monorepo does already give us efficiency gains in any contract that calls out to other contracts, since in each spot like that we call getAddressFor, and there we do compute them once before deployment in UsingRegistry.

👍

File: .circleci/config.yml [hunk 1]
```diff
@@ -69,7 +69,7 @@ jobs:
           command: |
             set -euo pipefail
             export CELO_MONOREPO_DIR="$PWD"
-            git clone --depth 1 https://github.com/celo-org/celo-monorepo.git ${CELO_MONOREPO_DIR} -b master
+            git clone --depth 1 https://github.com/celo-org/celo-monorepo.git ${CELO_MONOREPO_DIR} -b bricewang/registry-lookup
```
Comment (asaj) [inline .circleci/config.yml]: Same comment as on fixidity, you will only need to repoint the end-to-end tests for the first PR, and then the second PR can point to master

File: contract_comm/evm.go [hunk 2]
```diff
@@ -141,7 +141,7 @@ func MakeCallWithAddress(scAddress common.Address, abi abi.ABI, funcName string,
 	return executeEVMFunction(scAddress, abi, funcName, args, returnObj, gas, value, header, state, true)
 }
 
-func GetRegisteredAddress(registryId string, header *types.Header, state vm.StateDB) (*common.Address, error) {
+func GetRegisteredAddress(registryId [32]byte, header *types.Header, state vm.StateDB) (*common.Address, error) {
```
Comment (asaj) [inline contract_comm/evm.go]: nit: `registryId` vs `scRegistryId`

**Pull request merged**